pkg/labeler.go

```diff
@@ -68,6 +74,54 @@ func NewBranchCondition() Condition {
 	}
 }
 
+func NewFilesCondition(pr *gh.PullRequest) Condition {
+	ghToken := os.Getenv("GITHUB_TOKEN")
+	diffReq, _ := http.NewRequest("GET", pr.GetDiffURL(), nil)
+	diffReq.Header.Add("Authorization", "Bearer "+ghToken)
+	diffRes, _ := http.DefaultClient.Do(diffReq)
+
+	defer diffRes.Body.Close()
+
+	var diffRaw []byte
+	prFiles := make([]string, 0)
+	if diffRes.StatusCode == http.StatusOK {
+		diffRaw, _ = ioutil.ReadAll(diffRes.Body)
+
+		diff, _ := diffparser.Parse(string(diffRaw))
+		prFilesSet := map[string]struct{}{}
+		for _, file := range diff.Files {
+			prFilesSet[file.OrigName] = struct{}{}
+			prFilesSet[file.NewName] = struct{}{}
+		}
+		for k := range prFilesSet {
+			prFiles = append(prFiles, k)
+		}
+	}
+
+	return Condition{
+		GetName: func() string {
+			return "File matches regex"
+		},
+		Evaluate: func(pr *gh.PullRequest, matcher LabelMatcher) (bool, error) {
+			if len(matcher.Files) <= 0 {
+				return false, fmt.Errorf("files are not set in config")
+			}
+
+			log.Printf("Matching `%s` against: %s", strings.Join(matcher.Files, ", "), strings.Join(prFiles, ", "))
+			for _, file := range matcher.Files {
```

Comment:
Could we rename `file` here to `fileMatcher` or other similar word to ease reading the block below (e.g. it's easy to mistake `file` with an actual file).